torch/csrc/jit/script/init.cpp

```diff
@@ -7,7 +7,15 @@ namespace script {
 
 using ResolutionCallback = std::function<py::function(std::string)>;
 
-struct PythonResolver : public Resolver {
+// The visibility attribute is to avoid a warning about storing a field in the
+// struct that has a different visibility (from pybind) than the struct.
+#ifdef _WIN32
+#define VISIBILITY_HIDDEN
+#else
+#define VISIBILITY_HIDDEN __attribute__((visibility("hidden")))
+#endif
```

Comment:
We should put it in `utils` because it's a generally useful thing to have

Reply:
Or better compile with `-fvisibility=hidden` except for the structs we care about